validator/client/validator.go

```diff
@@ -120,6 +112,57 @@ func (v *validator) NextSlot() <-chan uint64 {
 	return v.ticker.C()
 }
 
+// LogValidatorGainsAndLosses logs important metrics related to this validator client's
+// responsibilities throughout the beacon chain's lifecycle. It logs absolute accrued rewards
+// and penalties over time, percentage gain/loss, and gives the end user a better idea
+// of how the validator performs with respect to the rest.
+func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64) error {
+	if slot%params.BeaconConfig().SlotsPerEpoch != 0 {
+		// Do nothing if we are not at the start of a new epoch.
+		return nil
+	}
+	epoch := slot / params.BeaconConfig().SlotsPerEpoch
+	if epoch == params.BeaconConfig().GenesisEpoch {
+		v.prevBalance = params.BeaconConfig().MaxDepositAmount
+	}
+	req := &pb.ValidatorPerformanceRequest{
+		Slot:      slot,
+		PublicKey: v.key.PublicKey.Marshal(),
+	}
+	resp, err := v.validatorClient.ValidatorPerformance(ctx, req)
+	if err != nil {
+		return err
+	}
+	newBalance := float64(resp.Balance) / float64(params.BeaconConfig().GweiPerEth)
+	log.WithFields(logrus.Fields{
+		"slot":  slot - params.BeaconConfig().GenesisSlot,
+		"epoch": (slot / params.BeaconConfig().SlotsPerEpoch) - params.BeaconConfig().GenesisEpoch,
```

Comment:
Looks a lot like helpers.SlotToEpoch

Reply:
Unfortunately it's in the beacon-chain package which is not public visibility. I don't want to move it to shared to make the PR bigger and introduce more changes that are unrelated in this PR